bin/strata-client/src/args.rs

```diff
@@ -1,171 +1,230 @@
+    /// Rpc host that the client will listen to.
+    #[argh(option, description = "rpc host")]
+    pub rpc_host: Option<String>,
 
-    #[argh(option, description = "bitcoind RPC user")]
-    pub bitcoind_user: Option<String>,
+    /// Rpc port that the client will listen to.
+    #[argh(option, description = "rpc port")]
+    pub rpc_port: Option<u16>,
 
-    #[argh(option, description = "bitcoind RPC password")]
-    pub bitcoind_password: Option<String>,
+    /// Other generic overrides to the [`Config`].
+    /// Will be used, for example, as `-o btcio.reader.client_poll_dur_ms=1000 -o exec.reth.rpc_url=http://reth`
+    #[argh(option, short = 'o', description = "generic config overrides")]
+    pub overrides: Vec<String>,
+}
 
-    /// Max retries for Bitcoin RPC calls.
-    #[argh(option, description = "max retries for bitcoin RPC (default: 3)")]
-    pub bitcoind_retry_count: Option<u8>,
+impl Args {
+    /// Overrides config. First overrides with the generic overrides passed via `-o` and then
+    /// overrides the result with some of the commonly passed args. Returns if config was overridden
+    /// or not.
+    pub fn override_config(&self, config: &mut Config) -> anyhow::Result<bool> {
+        // Override using -o params.
+        let mut overridden = self.override_generic(config)?;
+
+        // Override by explicitly parsed args like datadir, rpc_host and rpc_port.
+        if let Some(datadir) = &self.datadir {
+            config.client.datadir = datadir.into();
+            overridden = true
+        }
+        if let Some(rpc_host) = &self.rpc_host {
+            config.client.rpc_host = rpc_host.to_string();
+            overridden = true
+        }
+        if let Some(rpc_port) = &self.rpc_port {
+            config.client.rpc_port = *rpc_port;
+            overridden = true
+        }
+        Ok(overridden)
+    }
 
-    /// Timeout duration for btc request retries in ms. Defaults to `1000`.
-    #[argh(
-        option,
-        description = "max interval between bitcoin RPC retries in ms (default: 1000)"
-    )]
-    pub bitcoind_retry_interval: Option<u64>,
+    /// Override config using the generic overrides. The idea here is to first convert the
+    /// [`Config`] to json and then apply the overrides by splitting the key by a dot.
+    fn override_generic(&self, config: &mut Config) -> anyhow::Result<bool> {
+        let original = config.clone();
+        // Convert config as json
+        let mut json_config = to_value(&mut *config).context("Config json serialization failed")?;
 
-    #[argh(option, short = 'n', description = "L1 network to run on")]
-    pub network: Option<Network>,
+        for (path, val) in parse_overrides(&self.overrides)?.iter() {
+            apply_override(path, val, &mut json_config)?;
+        }
+        // Convert back to json.
+        *config = from_value(json_config)
+            .context("Should be able to create Config from serde json Value")?;
```

Comment:
These contexts are for adding more information about what the program was trying to do when the fn was called, not just literally what went wrong if there were to be an error.  That would be the job of the inner error type.